### PR TITLE
massive update:

### PR DESCRIPTION
**Pull request by sapphonie**
store connecting clients in a whitelist w/ipset using hex string searching combined with strict length checking of signon/signoff packets
ditch conntrack
use prerouting raw table for most of the rules

### CODE REVIEW

**Comment by sapphonie** [comment]
TODO: Add checking for clients still sending packets while "signed on"

File: ipt.sh [hunk 1]
```diff
@@ -4,42 +4,48 @@
...
 
-## =================================================================
-## CONFIGURATION
-## -----------------------------------------------------------------
 # this isnt just for fun, this lets me easily grep for the rules and delete and recreate them
 COMMENT="-m comment --comment="sappho.io""
 LOGPREFIX="[srcds-ipt]"
 # log up to every 30 seconds at max so we dont hog io
-LOGLIMIT="-m limit --limit 2/min"
+LOGLIMIT="-m limit --limit 1/min"
```
Comment (freylint) [inline ipt.sh:20]: Statics like this should be put in readonly var at the top of the file.
https://stackoverflow.com/questions/7023025/how-do-i-declare-a-constant-in-shell-script

File: ipt.sh [hunk 2]
```diff
@@ -4,42 +4,48 @@
...
-
+# default interface detection
 defaultin=$(route | grep '^default' | grep -o '[^ ]*$')
 
-ipt=""
-
+# ports setup
 ports="-m multiport --dports ${PORTMIN}:${PORTMAX} "
 
+# iptables command setup
```
Comment (freylint) [inline ipt.sh:36]: This appears to be behaving like a function.
Consider turning it into a subroutine name get_ipt or similar.

ie:
```bash
function get_ipt () {
    if [[ ${usedocker} == true ]]; then
    echo "Detected docker."
    return "iptables -I DOCKER-USER 1"
else
    echo "No docker.
    return "iptables -I INPUT 1"
}

readonly ipt=get_ipt()
```

File: ipt.sh [hunk 3]
```diff
@@ -48,126 +54,168 @@ else
...
+    -j LOG ${LOGLIMIT} --log-ip-options                                     \
     --log-prefix "${LOGPREFIX} len > 2521: "
 
+RULE_FILTER="-m string --algo bm --hex-string"
 
-## 3: UDP game packets
-## Allow "Established" packets so that we dont stomp on legit gamers
-## This rule goes last so it gets inserted first
-RULE_FILTER="-m state --state ESTABLISHED,RELATED"
+# 5: Log client signons
```
Comment (freylint) [inline ipt.sh]: Investigate if it's possible to do this async / as a different process to prevent io locking in the pre router.

File: ipt.sh [hunk 4]
```diff
@@ -48,126 +54,168 @@ else
...
+${ipt_pre_raw} -p udp -i ${defaultin} ${COMMENT}                            \
```
Comment (freylint) [inline ipt.sh:113]: A lot of duplicate code here. Consider turning into a function and calling it in a loop.

File: ipt.sh [hunk 5]
```diff
@@ -48,126 +54,168 @@ else
...
+# INPUT / DOCKER-USER
+#########################################################################
+
+
+RULE_FILTER="-m hashlimit --hashlimit-name speedlimit --hashlimit-mode srcip,dstport --hashlimit-above 2/sec --hashlimit-burst 4"
+
+# 3: UDP Spam
+# Should never see so much traffic from the same IP to the same port
+# ignore signed on and trusted users
+${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER}                            \
```
Comment (freylint) [inline ipt.sh]: Duplicate code that has potential to be turning into a function and loop.

**Review by freylint — CHANGES_REQUESTED** [review]
Code has potential for quality improvement. Can't verify functionality.

File: ipt.sh (hunk 3 above)
Comment (sapphonie) [reply ipt.sh:114]: dont care didnt ask plus youre gay

File: ipt.sh (hunk 2 above)
Comment (sapphonie) [reply ipt.sh:36]: i don't think this is using `return` properly, i think that it needs to be a numeric value only?

File: ipt.sh (hunk 3 above)
Comment (sapphonie) [reply ipt.sh]: it's netfilter, they worry about that, not me

**Pull request merged**